### DISCUSSION TITLE
Catch breakpoint event

### DISCUSSION THREAD

**Discussion by Nuthin2do** [CodinGame/monaco-vscode-api#503] [Q&A] [answered]
Hi. Can anybody tell me how i can catch the event when a breakpoint is set? 

For example: Previously i used the plain monaco-editor. There i registered an event for onMouseDown, checked if the clicked element's class is "glyph-margin-widgets" and then i rendered a breakpoint for this line and could save this breakpoint (with the lineNumber) in a List/Array to export to another application for example. 

And am I right in assuming that i need to register an Extension like in debugger.ts in the demo to be able to set breakpoints? 

Thanks for help in advance.

**Comment by CGNonofr** [comment]
I'll assume you are using the debugger service override.

What is you use-case that requires to have access to the breakpoint definitions from you code? You probably just better implement you debugger using the VSCode extension api and the Debug Adapter protocol

**Comment by Nuthin2do** [comment]
Hi, thanks for answering.

The use case is like that i have a C# application where the editor is embedded. I need the editor to write my scripts and being able to set breakpoints. The models and their breakpoints plus the code i send back to C#. So i need the list of breakpoints. If i implement the debugger to be able to set the breakpoints (if i understood you correctly) then i need to get the list where the breakpoints are stored in order to be able to react to that in C#. There should be something like that, right?! Because in the demo the breakpoints set are listed in the panel on the left. Can you tell me how or where i can get this list respectively where those breakpoints are stored?

Thank you very much.

**Comment by CGNonofr** [comment, marked as answer]
You can use `StandaloneServices.get(IDebugService).getModel().onDidChangeBreakpoints`
